lib/feedx/pusher.rb

```diff
@@ -54,9 +63,10 @@ def detect_compress(val)
     end
 
     def write_to(io)
-      stream = @format.new(io)
-      enum = @relation.respond_to?(:find_each) ? :find_each : :each
-      @relation.send(enum) {|rec| stream.write(rec) }
+      stream   = @format.new(io)
+      relation = @enum.is_a?(Proc) ? @enum.call : @enum
```

Comment:
I'd just check `@enum.respond_to?(:call)` (really doesn't matter, but would allow not only Proc passed but also some custom class/object with call method; though t makes sense only if `:enum` option is going to be kept).